crates/rover-error/src/metadata/mod.rs

```diff
@@ -0,0 +1,43 @@
+mod code;
+mod suggestion;
+
+use code::Code;
+use suggestion::Suggestion;
+
+use houston::HoustonProblem;
+use rover_client::RoverClientError;
+
+#[derive(Default, Debug)]
+pub struct Metadata {
+    pub suggestion: Option<Suggestion>,
+    pub code: Option<Code>,
+}
+
+impl From<&mut anyhow::Error> for Metadata {
+    fn from(error: &mut anyhow::Error) -> Self {
+        if let Some(rover_client_error) = error.downcast_ref::<RoverClientError>() {
```

Comment:
To get back to you about the `downcast_ref` we were talking about during the call yesterday: I think my idea was not good, since what I was thinking about was an abstraction that I'd maybe use in a publicly facing API, but we really should stick to more straightforward approach, like you have here. Downcasting is already a bit of a bleeding edge of rust, and abstracting on top of it will just give us a headache in the long run.

TL;DR: ignore my yesterday's downcast blunders.